proof/refine/AARCH64/ArchKHeap_R.thy

```diff
@@ -1165,835 +301,30 @@ lemma setObject_other_arch_corres:
-  apply (rule setObject_ifunsafe'[where P="\<top>", simplified])
-    apply (clarsimp simp: updateObject_default_def in_monad
-                  intro!: equals0I)+
-  apply (simp add: setObject_def split_def)
-  apply (wp updateObject_default_inv | simp)+
-  done
-
-lemma setEndpoint_idle'[wp]:
-  "\<lbrace>\<lambda>s. valid_idle' s\<rbrace>
-   setEndpoint p v
-   \<lbrace>\<lambda>_. valid_idle'\<rbrace>"
-  unfolding setEndpoint_def
-  apply (wp setObject_idle')
-       apply (simp add: objBits_simps' updateObject_default_inv idle_tcb_ps_def)+
-  done
-
-crunch setEndpoint
-  for it[wp]: "\<lambda>s. P (ksIdleThread s)"
-  (simp: updateObject_default_inv)
-
-lemma setObject_ksPSpace_only:
-  "\<lbrakk> \<And>p q n ko. \<lbrace>P\<rbrace> updateObject val p q n ko \<lbrace>\<lambda>rv. P \<rbrace>;
-        \<And>f s. P (ksPSpace_update f s) = P s \<rbrakk>
-     \<Longrightarrow> \<lbrace>P\<rbrace> setObject ptr val \<lbrace>\<lambda>rv. P\<rbrace>"
-  apply (simp add: setObject_def split_def)
-  apply (wp | simp | assumption)+
-  done
-
-lemma setObject_ksMachine:
-  "\<lbrakk> \<And>p q n ko. \<lbrace>\<lambda>s. P (ksMachineState s)\<rbrace> updateObject val p q n ko \<lbrace>\<lambda>rv s. P (ksMachineState s)\<rbrace> \<rbrakk>
-     \<Longrightarrow> \<lbrace>\<lambda>s. P (ksMachineState s)\<rbrace> setObject ptr val \<lbrace>\<lambda>rv s. P (ksMachineState s)\<rbrace>"
-  by (simp add: setObject_ksPSpace_only)
-
-lemma setObject_ksInterrupt:
-  "\<lbrakk> \<And>p q n ko. \<lbrace>\<lambda>s. P (ksInterruptState s)\<rbrace> updateObject val p q n ko \<lbrace>\<lambda>rv s. P (ksInterruptState s)\<rbrace> \<rbrakk>
-     \<Longrightarrow> \<lbrace>\<lambda>s. P (ksInterruptState s)\<rbrace> setObject ptr val \<lbrace>\<lambda>rv s. P (ksInterruptState s)\<rbrace>"
-  by (simp add: setObject_ksPSpace_only)
-
-lemma valid_irq_handlers_lift':
-  assumes x: "\<And>P. \<lbrace>\<lambda>s. P (cteCaps_of s)\<rbrace> f \<lbrace>\<lambda>rv s. P (cteCaps_of s)\<rbrace>"
-  assumes y: "\<And>P. \<lbrace>\<lambda>s. P (ksInterruptState s)\<rbrace> f \<lbrace>\<lambda>rv s. P (ksInterruptState s)\<rbrace>"
-  shows      "\<lbrace>valid_irq_handlers'\<rbrace> f \<lbrace>\<lambda>rv. valid_irq_handlers'\<rbrace>"
-  apply (simp add: valid_irq_handlers'_def irq_issued'_def)
-  apply (rule hoare_use_eq [where f=cteCaps_of, OF x y])
-  done
-
-lemmas valid_irq_handlers_lift'' = valid_irq_handlers_lift' [unfolded cteCaps_of_def]
-
-crunch setEndpoint
-  for ksInterruptState[wp]: "\<lambda>s. P (ksInterruptState s)"
-  (wp: setObject_ksInterrupt updateObject_default_inv)
-
-lemmas setEndpoint_irq_handlers[wp]
-    = valid_irq_handlers_lift'' [OF set_ep_ctes_of setEndpoint_ksInterruptState]
-
-declare set_ep_arch' [wp]
+(* FIXME arch-split: check with other arches what is actually in common, standardise *)
+lemma setEndpoint_pspace_in_kernel_mappings'[KHeap_R_assms]:
+  "setEndpoint p ko \<lbrace>pspace_in_kernel_mappings'\<rbrace>"
```

Comment:
Should this happen now or is it intended for when more progress is made into Refine?

Reply:
Ah, yes indeed. It seems that on all arches we can say that `setObject` preserves `pspace_in_kernel_mappings'`, so we can try to export that instead and maybe even do the Endpoint/Notification instantiations outside of Arch if RISCV64/X64 permit. Will take another look.

Reply:
this should be resolved now, but see the setObject turbulence in latest commits